### PR TITLE
8272805: Avoid looking up standard charsets

### PR DESCRIPTION
**Pull request by mrserb**
This is the continuation of JDK-8233884, JDK-8271456, and JDK-8272120.

In many places standard charsets are looked up via their names, for example:
absolutePath.getBytes("UTF-8");

This could be done more efficiently(up to x20 time faster) with use of java.nio.charset.StandardCharsets:
absolutePath.getBytes(StandardCharsets.UTF_8);

The later variant also makes the code cleaner, as it is known not to throw UnsupportedEncodingException in contrary to the former variant.

This change includes:
 * demo/utils
 * jdk.xx packages
 * Some places were missed in the previous changes. I have found it by tracing the calls to the Charset.forName() by executing tier1,2,3 and desktop tests.

Some performance discussion: https://github.com/openjdk/jdk/pull/5063

Code excluded in this fix: the Xerces library(should be fixed upstream), J2DBench(should be compatible to 1.4), some code in the network(the change there are not straightforward, will do it later).

Tested by the tier1/tier2/tier3 tests on Linux/Windows/macOS.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272805](https://bugs.openjdk.java.net/browse/JDK-8272805): Avoid looking up standard charsets

### Reviewers
 * [Weijun Wang](https://openjdk.java.net/census#weijun) (@wangweij - **Reviewer**) ⚠️ Review applies to 2d9c80b838dd0d5bec8eb996c465126a108dc145
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to e7127644d8a947cf43fbbde04e5fe0e4696f9dc3
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**) ⚠️ Review applies to e7127644d8a947cf43fbbde04e5fe0e4696f9dc3
 * @turbanoff (no known github.com user name / role) ⚠️ Review applies to e7127644d8a947cf43fbbde04e5fe0e4696f9dc3
 * [Alexander Zvegintsev](https://openjdk.java.net/census#azvegint) (@azvegint - **Reviewer**) ⚠️ Review applies to e7127644d8a947cf43fbbde04e5fe0e4696f9dc3
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to e7127644d8a947cf43fbbde04e5fe0e4696f9dc3

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5210/head:pull/5210` \
`$ git checkout pull/5210`

Update a local copy of the PR: \
`$ git checkout pull/5210` \
`$ git pull https://git.openjdk.java.net/jdk pull/5210/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5210`

View PR using the GUI difftool: \
`$ git pr show -t 5210`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5210.diff">https://git.openjdk.java.net/jdk/pull/5210.diff</a>

</details>

### CODE REVIEW

**Review by wangweij — APPROVED** [review]
The security related change looks fine to me.

File: src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java [hunk 1]
```diff
@@ -343,7 +339,7 @@ private int getNamespacePid(int pid) throws AttachNotSupportedException, IOExcep
         Path statusPath = Paths.get(statusFile);
 
         try {
-            for (String line : Files.readAllLines(statusPath, StandardCharsets.UTF_8)) {
+            for (String line : Files.readAllLines(statusPath, UTF_8)) {
```
Comment (AlanBateman) [inline src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java:342]: The 1-arg readAllLines is specified to use UTF-8 so you can drop the second parameter here if you want.

**Comment by turbanoff** [comment]
I think it's worth to update _static_ initializer in `sun.datatransfer.DataFlavorUtil.CharsetComparator` too.
изображение

**Review by naotoj — APPROVED** [review]
Looks good.

**Review by dfuch — APPROVED** [review]
Changes to http server look good to me.

File: src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java (hunk 1 above)
Comment (mrserb) [reply src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java]: Thank you for the suggestion, I have fixed this and a few other places as well.

**Comment by mrserb** [comment]
> I think it's worth to update _static_ initializer in `sun.datatransfer.DataFlavorUtil.CharsetComparator` too.

Updated as suggested.

**Review by turbanoff — APPROVED** [review]

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**